_source/_posts/2021-11-12-native-sso.md

```diff
@@ -0,0 +1,233 @@
+- **Login App 1 through the embedded browser**. The login session is saved in a persistent cookie.
+- **Login App 2 through the embedded browser**. Since the login session is still in the persistent cookie, the user is not prompted for login.
+
+## Native SSO based on token exchange
+
+Okta's Native SSO solution is based on token exchange, and it builds on an OIDC draft spec [Native SSO for Mobile Apps](https://openid.net/specs/openid-connect-native-sso-1_0.html). The following diagrams, taken from the spec, illustrate the flow to enable Native SSO.
+
+Sign-in to the first application is similar to a normal OIDC sign in, using a system browser. The only difference is that we request the `device_SSO` scope, and in return, we receive a `device_secret` along with the returned tokens.
+
+{% img blog/native-sso/native-app-1-interactions.png alt:"Native App 1 Login interaction diagram" width:"600" %}{: .center-image }
+
+When Native App 2 requests login, it can simply take the ID token and Device Secret returned for App 1, and call the token endpoint to exchange for tokens for App 2. The returned tokens for App 2 have their own refresh lifecycle.
+
+{% img blog/native-sso/native-app-2-interactions.png alt:"Native App 2 Login interaction diagram" width:"600" %}{: .center-image }
+
+Our [Native SSO Guide](https://developer.okta.com/docs/guides/configure-native-sso/main/) goes into more detail on how to enable Native SSO, and demonstrates the interaction through curl commands.
+
+## How is it different
+
+The token-exchange-based Native SSO solution is simpler and better than sharing Web SSO in at least several regards:
+- **Out-of-the-box config**: Sharing web SSO requires you to use an API to configure all users to use a persistent cookie for session management (the default is to use a session cookie). A login session will persist even after you close the browser, which may be a surprise for some users. In contrast, Native SSO is one of the supported grant types that can be easily enabled in the Okta Admin Console.
+- **Single LogOut**: It is not possible to single logout all applications with the web SSO sharing solution. If you want to logout from all apps, you have to logout from each app one by one. In contrast, in Native SSO, one can simply revoke `device_secret`, then all apps automatically sign out.
+- **Multiple devices**: Sharing web SSO can only work on the same device. In contrast, Native SSO can work across devices, as long as `device_secret` can be safely shared. In fact, the remaining portion of this blog demonstrates how to implement Native SSO across multiple Apple devices.
+
+## Single Sign On Across Multiple Apple Devices
+
+We all own multiple devices such as iPhone, Macbook, iPad. It would be great to *single sign on* to apps across all the devices we own. In this section, we demonstrate how to sign into a mobile app on an iOS device and a desktop app on a Macbook by leveraging iCloud keychain.
```

Comment:
```suggestion
We all own multiple devices such as iPhone, MacBook, or iPad. It would be great to *single sign on* to apps across all the devices we own. In this section, we demonstrate how to sign into a mobile app on an iOS device and a desktop app on a Macbook by leveraging iCloud keychain.
```